apps/ensnode/src/api/index.ts

```diff
@@ -1,16 +1,130 @@
-import { db } from "ponder:api";
+import { db, publicClients } from "ponder:api";
 import schema from "ponder:schema";
+import { ponderMetadata } from "@ensnode/ponder-metadata/middleware";
 import { graphql as subgraphGraphQL } from "@ensnode/ponder-subgraph/middleware";
+import { swaggerUI } from "@hono/swagger-ui";
 import { Hono } from "hono";
+import { describeRoute, openAPISpecs } from "hono-openapi";
+import { resolver, validator as zValidator } from "hono-openapi/zod";
+import { cors } from "hono/cors";
 import { client, graphql as ponderGraphQL } from "ponder";
+import { z } from "zod";
+// For extending the Zod schema with OpenAPI properties
+import "zod-openapi/extend";
+import { getEnsDeploymentChain } from "../lib/ponder-helpers";
 
 const app = new Hono();
 
+// use CORS middleware
+app.use(
+  cors({
+    origin: "*",
+  }),
+);
+
+const packageJson = await import("../../package.json").then((m) => m.default);
+const OPEN_API_MANIFEST_PATH = "/api/openapi.json";
+
+app.get(
+  OPEN_API_MANIFEST_PATH,
+  openAPISpecs(app, {
+    documentation: {
+      info: {
+        title: packageJson.name,
+        version: packageJson.version,
+        description: packageJson.description,
+      },
+      servers: [
+        {
+          url: process.env.PUBLIC_SERVICE_URL,
+        },
+      ],
+    },
+  }),
+);
+
+// Use the middleware to serve Swagger UI at /ui
+app.get("/swagger", swaggerUI({ url: OPEN_API_MANIFEST_PATH }));
+
+// use ENSNode middleware at /metadata
+app.get(
+  "/metadata",
+  describeRoute({
+    description: "Metadata describing the ENSNode service instance",
+    responses: {
+      200: {
+        description: "Metadata object",
+        content: {
+          "text/json": {
+            schema: resolver(
+              z.any().openapi({
+                type: "object",
+              }),
+            ),
+          },
+        },
+      },
+    },
+  }),
+
+  ponderMetadata({
+    app: {
+      name: packageJson.name,
+      version: packageJson.version,
+    },
+    env: {
+      ACTIVE_PLUGINS: process.env.ACTIVE_PLUGINS,
+      DATABASE_SCHEMA: process.env.DATABASE_SCHEMA,
+      ENS_DEPLOYMENT_CHAIN: getEnsDeploymentChain(),
+    },
+    db,
+    fetchPrometheusMetrics: () =>
+      fetch(`http://localhost:${process.env.PORT}/metrics`).then((res) => res.text()),
```

Comment:
Ponder does not offer any internal API to read metrics, so I'm using its public API over localhost network to read metrics.

@lightwalker-eth perhaps when chatting with Ponder team you could highlight a need for us to be able to use some ponder internal APIs.